uns_plugin/uns_input_processor.go

```
@@ -71,9 +73,20 @@ func (p *MessageProcessor) ProcessRecord(record *kgo.Record) *service.Message {
 
 	msg := service.NewMessage(record.Value)
 
-	// Add headers to the meta field if present
+	// Add headers to the meta field if present, converting based on metadataFormat
 	for _, h := range record.Headers {
-		msg.MetaSetMut(h.Key, h.Value)
+		switch p.metadataFormat {
+		case MetadataFormatString:
+			msg.MetaSetMut(h.Key, string(h.Value))
+		case MetadataFormatBytes:
+			msg.MetaSetMut(h.Key, h.Value)
+		default:
+			// Unreachable: The configuration parser should only accept valid
+			// values for the MetadataFormat enum. This ensures that if a
+			// new MetadataFormat variant gets added we also update this
+			// part of the code.
+			panic(fmt.Sprintf("Unknown MetadataFormat: %#v", p.metadataFormat))
```

are we also sure to panic here instead of using 1 as a default? (although it states unreachable)